es-vue-base/src/lib-components/EsFileUpload.vue

```diff
@@ -0,0 +1,230 @@
+            :state="Boolean(pickedItems)"
+            :accept="fileTypesAsString"
+            style="display: none"
+            multiple />
+    </div>
+</template>
+
+<script lang="js">
+import { BFormFile } from 'bootstrap-vue';
+import EsButton from './EsButton.vue';
+import IconUpload from '../lib-icons/icon-upload.vue';
+
+export default {
+    name: 'EsFileUpload',
+    components: {
+        BFormFile,
+        EsButton,
+        IconUpload,
+    },
+    props: {
+        uploadUrls: {
+            type: Array,
+            default: () => [],
+        },
+        fileTypes: {
+            type: Array,
+            default: () => [],
+        },
+        maxFileSize: {
+            type: Number,
+            default: 100,
+            required: false,
+        },
+        uploadSuccessStatusCode: {
+            type: Number,
+            default: 200,
+            required: false,
+        },
+        collapsed: {
+            type: Boolean,
+            default: false,
+            required: false,
+        },
+    },
+    data() {
+        return {
+            pickedItems: [],
+            files: [],
+            fileUrls: [], // For thumbnail preview
+            active: false,
+        };
+    },
+    computed: {
+        fileTypesAsString() {
+            return this.fileTypes.join(', ');
+        },
+    },
+    watch: {
+        uploadUrls: {
+            handler(newUrls) {
+                if (newUrls.length > 0 && newUrls.length === this.files.length) this.upload();
+            },
+            deep: true,
+        },
+        files: {
+            handler(newFiles) {
+                if (newFiles.length > 0 && newFiles.length === this.uploadUrls.length) this.upload();
+            },
+            deep: true,
+        },
+        pickedItems(newVal) {
+            this.files = newVal;
+            if (newVal.length > 0 && this.fileSizeValid()) {
+                this.$emit('readyToUpload', this.files.length);
+                this.readFilesIntoUrl(newVal);
+            }
+        },
+    },
+    methods: {
+        fileSizeValid() {
+            // Check if the combined file sizes exceed the maxFileSize prop
+            const totalFileSize = this.files.reduce((acc, file) => acc + file.size, 0);
+            if (totalFileSize > this.maxFileSize * 1000000) {
+                this.$emit('fileSizeError');
+                return false;
+            }
+            return true;
+        },
+        readFilesIntoUrl(files) {
+            this.fileUrls = [];
+            files
+                .forEach((file) => {
+                    const fileReader = new FileReader();
+                    fileReader.onload = () => {
+                        this.$emit('fileDataRead', {
+                            name: file.name, type: file.type, size: file.size, data: fileReader.result,
+                        });
+                    };
+                    fileReader.readAsDataURL(file);
+                });
+        },
```

Comment:
Fine to emit these details but this component should be responsible for displaying the details to the user. I would have expected to see my file's details in the component somewhere when this is triggered.